src/libraries/System.Text.Json/docs/ReferenceHandling_spec.md

```diff
@@ -0,0 +1,810 @@
+        }
+    }";
+```
+On System.Text.Json:
+```cs
+public static void ReadJsonWithPreservedReferences(){
+    var options = new JsonSerializerOptions
+    {
+        ReferenceHandling = ReferenceHandling.Preserve
+    };
+
+    Employee angela = JsonSerializer.Deserialize<Employee>(json, options);
+    Console.WriteLine(object.ReferenceEquals(angela, angela.Manager.Subordinates[0])); //prints: true.
+}
+```
+
+On Newtonsoft.Json:
+```cs
+public static void ReadJsonWithPreservedReferences(){
+    var options = new JsonSerializerSettings
+    {
+        //Newtonsoft.Json reads metadata by default, just setting the option for ilustrative purposes.
+        MetadataPropertyHanding = MetadataPropertyHandling.Default 
+    };
+
+    Employee angela = JsonConvert.DeserializeObject<Employee>(json, settings);
+    Console.WriteLine(object.ReferenceEquals(angela, angela.Manager.Subordinates[0])); //prints: true.
+}
+```
+
+# Other languages
+
+## Newtonsoft.Json
+Newtonsoft.Json contains settings that you can enable to deal with such problems.
+* For Serialization:
+  * [`ReferenceLoopHandling`](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_ReferenceLoopHandling.htm)
+  * [`PreserveReferencesHandling`](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_PreserveReferencesHandling.htm)
+* For Deserialization:
+  * [`MetadataPropertyHandling`](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_MetadataPropertyHandling.htm)
+
+When using `ReferenceLoopHandling.Ignore`, other objects that were already seen on the current graph branch will be ignored on serialization.
+
+When using `PreserveReferencesHandling.All` you are signaling that your resulting JSON will contain *metadata* properties `$ref`, `$id` and `$values` which are going to act as reference identifiers (`$id`) and pointers (`$ref`). 
+Now, to read back those references, you have to use `MetadataPropertyHandling.Default` to indicate that *metadata* is expected in the payload passed to the `Deserialize` method.
+
+* Pros
+  * If we opt-in for this we could provide compatibility with Newtonsoft which is always desired by the community.
+* Cons
+  * Quite invasive, (it affects `JsonException.Path`, `JsonSerializerOptions.IgnoreNullValues`, `JsonPropertyNameAttribute`, and Converters).
+  * This would break existing converters i.e: an array converter may expect the first token to be "[" and a preserved array starts with "{".
+    * perhaps converters are more feasible with the JSON path implementation.
+  * We will now accept that an array comes in valid format when starts with a curly brace "{"; below issue is related to guard against NRE when this happens:
+    * https://github.com/dotnet/corefx/issues/41839
+
+## dojo toolkit (JavaScript framework)
+https://dojotoolkit.org/reference-guide/1.10/dojox/json/ref.html
+
+Similar: https://www.npmjs.com/package/json-cyclic 
+
+* id-based (ignore this approach since is the same the one of Newtonsoft.Json)
```

Comment:
> ignore this approach since is the same the one of Newtonsoft.Json

What do you mean by this?

Is it the same as Newtonsoft OR different? Did you mean this?
```suggestion
* id-based (ignore this approach since it is the same as `Newtonsoft.Json`)
```